### PR TITLE
Enable dump-skp-on-shader-compilation in drive

### PR DESCRIPTION
**Pull request by liyuqian**
Previously, it's only available in run. There's no reason not to make it available in drive too.

### CODE REVIEW

**Comment by jonahwilliams** [comment]
There are some test cases in https://github.com/flutter/flutter/blob/master/packages/flutter_tools/test/commands.shard/hermetic/drive_test.dart#L336, could they be expanded to verify this flag?

**Comment by liyuqian** [comment]
@jonahwilliams : tests added.

File: packages/flutter_tools/test/commands.shard/hermetic/drive_test.dart [hunk 1]
```diff
@@ -459,6 +459,116 @@ void main() {
...
+        when(mockLaunchResult.started).thenReturn(true);
+        when(mockDevice.startApp(
+          null,
+          mainPath: anyNamed('mainPath'),
+          route: anyNamed('route'),
+          debuggingOptions: anyNamed('debuggingOptions'),
+          platformArgs: anyNamed('platformArgs'),
+          prebuiltApplication: anyNamed('prebuiltApplication'),
+        )).thenAnswer((Invocation invocation) async {
+          debuggingOptions = invocation.namedArguments[kDebugOptionsSymbol];
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/commands.shard/hermetic/drive_test.dart]: nit: use the literal form of Symbol
```suggestion
          debuggingOptions = invocation.namedArguments[#debuggingOptions];
```

File: packages/flutter_tools/test/commands.shard/hermetic/drive_test.dart [hunk 2]
```diff
@@ -459,6 +459,116 @@ void main() {
...
+          return mockLaunchResult;
+        });
+        when(mockDevice.isAppInstalled(any)).thenAnswer((_) =>
+        Future<bool>.value(false));
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/commands.shard/hermetic/drive_test.dart]: Formatting here is a bit weird, and the type needs to be specified.

```
when(mockDevice.isAppInstalled(any))
  .thenAnswer((Invocation _) => Future<bool>.value(false));
```

File: packages/flutter_tools/test/commands.shard/hermetic/drive_test.dart [hunk 3]
```diff
@@ -459,6 +459,116 @@ void main() {
...
+
+        testApp = fs.path.join(tempDir.path, 'test', 'e2e.dart');
+        testFile = fs.path.join(tempDir.path, 'test_driver', 'e2e_test.dart');
+
+        testRunner = (List<String> testArgs, String observatoryUri) async {
+          throwToolExit(null, exitCode: 123);
+        };
+        appStopper = expectAsync1(
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/commands.shard/hermetic/drive_test.dart]: Here and below, indent with 2 spaces

**Review by jonahwilliams — APPROVED** [review]
LGTM once comments are addressed.

**Comment by Hixie** [comment]
@liyuqian sorry about that, you'll want to change your ProcessManager constructor call to ProcessManager.any.

**Comment by liyuqian** [comment]
Thank you @Hixie for the info! Just uploaded https://github.com/flutter/flutter/pull/43455

**Pull request merged**